app/scripts/components/analysis/define/use-custom-aoi.ts

```diff
@@ -75,52 +81,83 @@ function useCustomAoI() {
           return;
         }
       }
-      let feature: Feature = geojson.features[0];
-      if (!geojson.features || !geojson.features.length) {
+
+      if (!geojson?.features?.length) {
         setError('Error uploading file: Invalid GeoJSON');
         return;
       }
 
       let warnings: string[] = [];
-      if (geojson.features.length > 1) {
-        warnings = [
-          ...warnings,
-          'Files containing multiple features are not supported. Only the first feature will be used.'
-        ];
-      }
 
-      if (feature.geometry.type === 'MultiPolygon') {
-        warnings = [
-          ...warnings,
-          'Files containing multiple polygons are not supported. Only the first polygon will be used.'
-        ];
-        feature = multiPolygonToPolygon(feature as Feature<MultiPolygon>);
-      } else if (feature.geometry.type !== 'Polygon') {
+      if (
+        geojson.features.some(
+          (feature) =>
+            !['MultiPolygon', 'Polygon'].includes(feature.geometry.type)
+        )
+      ) {
         setError(
           'Wrong geometry type. Only polygons or multi polygons are accepted.'
         );
         return;
       }
 
-      const originalNumPoints = getNumPoints(feature as Feature<Polygon>);
-      let numPoints = originalNumPoints;
+      const features: Feature<Polygon>[] = geojson.features.reduce(
+        (acc, feature: Feature<Polygon | MultiPolygon>) => {
+          if (feature.geometry.type === 'MultiPolygon') {
+            return acc.concat(
+              multiPolygonToPolygons(feature as Feature<MultiPolygon>)
+            );
+          }
+
+          return acc.concat(feature);
+        },
+        []
+      );
+
+      if (features.length > 200) {
+        setError('Only files with up to 200 polygons are accepted.');
+        return;
+      }
+
+      // Simplify features;
+      const originalTotalFeaturePoints = features.reduce(
+        (acc, f) => acc + getNumPoints(f),
+        0
+      );
+      let numPoints = originalTotalFeaturePoints;
       let tolerance = 0.001;
-      while (numPoints > 2000) {
-        feature = simplifyFeature(feature as Feature<Polygon>, tolerance);
-        numPoints = getNumPoints(feature as Feature<Polygon>);
+      let simplifiedFeatures = [...features];
+
+      while (numPoints > 200 && tolerance < 5) {
+        simplifiedFeatures = simplifiedFeatures.map((feature) =>
+          simplifyFeature(feature, tolerance)
```

Comment:
I tested with this file (state parks in Hawaii): https://gist.github.com/hanbyul-here/fd587e3cb106a3458b39f78adb69249f , I got a critical error.

I think `simplifyFeature` simplifies polygons to two coordinates level (line), and made `union` function impossible. It might make sense to validate simplified polygons.

Reply:
@hanbyul-here You found something very interesting here. That is exactly what was happening and I fixed it by checking that each polygon doesn't get below 4 points.
The other thing that I saw with this dataset is that some of the polygon had rings (holes). That is not something currently supported by the editor, so I added some functionality to remove the holes and warn the user.

<img width="816" alt="image" src="https://user-images.githubusercontent.com/1090606/215834629-cd65965d-a935-4e55-b1a2-a2879f5a80d1.png">

Even though I don't think this dataset represent a real use case (areas are very small) it was great to use it to test!

@nerik The fact that each feature may be simplified to the point of becoming invalid is really problematic. For now this works, but we should look more closely at that other library you suggested.

Reply:
Yeah, I agree that the dataset is unlikely to be used for the dashboard. 
Thanks for handling the issue! Since the editor can't offer 'ring' type polygons, isn't it better to allow users to upload that type of polygons through the upload modal?

Reply:
The idea is that the different options lead to the same outcome. You'd not be able to edit those polygons properly with the editor, so I don't think we should allow them.
Not sure if having rings would even be a use case when performing analysis.